mlflow/utils/requirements_utils.py

```diff
@@ -445,6 +445,12 @@ def _check_requirement_satisfied(requirement_str):
     req = pkg_resources.Requirement.parse(requirement_str)
     pkg_name = req.name
 
+    if pkg_name == 'pyspark' and is_in_databricks_runtime():
+        # for databricks runtime pyspark, ignore micro-version mismatch.
+        # replace == specifier to be ~= specifier, but keep === specifier unchanged.
+        requirement_str = re.sub('([^=])==([^=])', r'\1~=\2', requirement_str)
```

Comment:
Can we run this operation after `_init_packages_to_modules_map`?

Reply:
```suggestion
        requirement_str = re.sub(r'==([^=])', r'~=\1', requirement_str)
```

Can we only replace the version specifier?

Reply:
This seems does not work for `===`
`re.sub(r'==([^=])', r'~=\1', 'abc===def')` get `'abc=~=def'`

Reply:
Got it, I misunderstood `===` was a typo, but it's a valid operator.